### ISSUE TITLE
DateTimePicker: AM/PM switcher is not screen reader accessible

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#61163]
## What problem does this address?

The AM/PM toggle that appears when `is12Hour={ true }` only expresses state through visual styling, and cannot be distinguished by screen readers.

<img width="202" alt="AM/PM switcher" src="https://github.com/WordPress/gutenberg/assets/555336/1902dc29-62fe-43b5-a927-7ce817944b39">

## What is your proposed solution?

A similarly styled component, [`RadioGroup`](https://wordpress.github.io/gutenberg/?path=/docs/components-radiogroup--docs), is already deprecated in favor of [`ToggleGroupControl`](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-togglegroupcontrol--docs). Since we don't really use this UI pattern anymore, can we just replace it with a ToggleGroupControl? @WordPress/gutenberg-design

**Comment by jasmussen** [comment]
ToggleGroupControl seems perfect for this case. Would the compact size prop work there? Not a blocker, but can likely be useful in this context.

**Comment by jameskoster** [comment]
Probably a dumb question and/or rabbit hole, but does this need to be a separate control? Curious whether the time input could use a combobox pattern. This would allow you to select am/pm at the same time you choose the hour, and enable selection-by-mouse.

<img width="1041" alt="time" src="https://github.com/WordPress/gutenberg/assets/846565/ee90c5ef-804a-42db-b5f7-07632ba42d66">

**Comment by mirka** [comment]
> does this need to be a separate control?

Good point, I see how this can be enhanced to use a ComboboxControl. We can do a quick accessibility fix now by using a ToggleGroupControl for the AM/PM toggle, and once we have a rewritten ComboboxControlV2 it might be worth enhancing the inputs.

**Comment by patil-vipul** [comment]
Raised a PR for this issue here https://github.com/WordPress/gutenberg/pull/61562
@mirka @jasmussen

**Comment by MadGanGithub** [comment]
Hey @mirka, If this issue is open, I would like to work on this

**Comment by mirka** [comment]
@MadGanGithub It seems like @patil-vipul abandoned the PR (?), so please feel free to submit a new PR.

**Comment by patil-vipul** [comment]
@mirka I'll be pushing new PR with same changes, as the previous PR got messed up.

**Comment by ciampo** [comment]
The previous attempt https://github.com/WordPress/gutenberg/pull/61562 was closed without merging — @MadGanGithub would you be up for working on it?

**Comment by mirka** [comment]
We need to land this in the next month or two for this to be included in [WP 6.7](https://make.wordpress.org/core/6-7/). Do we still have any takers to work on this?

**Comment by ciampo** [comment]
Since it's been a while and no one unfortunately volunteered, here is a PR closing this issue #64800
